### DISCUSSION TITLE
About getter generates new instances

### DISCUSSION THREAD

**Discussion by minsungkim-qraft** [PyO3/pyo3#3937] [Questions] [answered]
Let's consider following code.

```rust
#[pyclass]
pub struct Container {
    #[pyo3(get)]
    pub internal: Vec<i64>,
}

#[pymethods]
impl Container {
    #[staticmethod]
    pub fn dummy() -> PyResult<Self> {
        Ok(Container { internal: (0..10).collect() })
    }
}
```

and imagine if you try to manipulate `Container::internal` in Python.

```python
container = Container.dummy()
print(container.internal)
container.internal.append(10)
print(container.internal)
```

and if you run the code, you will see identical two lines on the terminal. In other words, 10 is not appended.

1. Is this identical to [this troubleshooting issue](https://pyo3.rs/v0.20.3/faq.html?highlight=clone#pyo3get-clones-my-field)?
2. Is this also same for Rust enums decorated by `pyclass`? In Python, every Enum instance is unique, that is, `a is b` is always true if `a` and `b` are same enum object. However, if PyO3 clones same enum objects then this might be false, so I wondered. If PyO3 is cloning same enum object(therefore `a is b == False`), could you please elaborate why?

**Comment by davidhewitt** [comment, marked as answer]
Sorry for the slow reply. In answer to your questions:

1. Yes, it is equivalent to the FAQ issue.
2. At the moment PyO3 does not create enum singletons but that is actively being worked on. See #3059.
